pkg/util/testing/not_found_error_cmp.go

```diff
@@ -75,3 +75,27 @@ func (matcher *isForbiddenErrorMatch) FailureMessage(actual interface{}) (messag
 func (matcher *isForbiddenErrorMatch) NegatedFailureMessage(actual interface{}) (message string) {
 	return format.Message(actual, "not to be Forbidden error")
 }
+
+func BeInvalidError() types.GomegaMatcher {
+	return &isInvalidErrorMatch{}
```

Comment:
Maybe we can have a single class that implements all the matchers by accepting one of the apierrors functions and a name.